react/week5/github-single-usersearch-app/src/Components/SearchResults.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import SearchResult from './SearchResult';
+import { Link } from 'react-router-dom';
+
+const SearchResults = ({ results }) => {
+    const style = {
+        display: 'flex',
+        flexWrap: 'wrap',
+        alignItems: 'stretch',
+        justifyContent: 'center',
+        height: '100%'
+    }
+    return (
+        <div style={style}>
+            {
+                results.length === 0 ? (<p>No result</p>)
```

Comment:
How long would it take you to understand this piece of code in a couple of weeks? Not long, but you would still have to think. Try to image that all code you write should simply explain what it does without the reader having to do any thinking at all.

Reply:
I am not sure how to change it .Do you thinks it would be easier if I write it like this 
{ if (result.length ===0){
return (<p> No result </p>)
}else{...}}

Reply:
Well, first of all you should put your css in a css file where it belongs. Secondly you should be very careful when naming things. When you write a small program that is not meant to be seen by others and never is intended to grow any further, then these things don't really matter, which, granted, might be the case here. 

But in most cases naming becomes critical in the long run. Same with principles such as separation of concerns like keeping your css and your logic separated. Another important aspect of programming is the idea of creating abstractions for your code. This will transform your code from an imperative style into a declarative and readable style. When we begin to code in a declarative style that seeks to descibe to the reader what the code does instead of how the code works, we begin to organize our code and create meaningful structures that is easier to understand and change later on. 
The four pillars of object oriented programming are :
Inheritance, Polymorphism, Encapsulation and Abstraction. Do you what they mean and how to apply them?
What about the SOLID principles? 
Single responsibility pattern
Opened/closed principle
Liskov substitution
Interface segragation
Dependency Inversion.

We are not really concerned with getting things to "work" as programmers. This should and can always be accomplished on a single day. Regardless of what we are dealing with. 
A programmer begins his work on the second day. Restructuring, renaming, dividing, creating abstractions, extracting, testing, redefining the architecture... until the code can be read without thinking and everything is modular and decomposed into separate isolated and testable units. The better the programmer the more readable the code becomes. A good programmer never obscures his intentions. Everything is well defined and organised. The simpler the better. 
A beginner, on the other hand, writes complicated hard to read code that is not clear in its intentions, with generic names and large blocks of code that must be executed by the reader to be understood.